UbuntuDrivers/detect.py

```diff
@@ -70,58 +70,91 @@ def system_modaliases(sys_path=None):
     return aliases
 
 
-def _check_video_abi_compat(apt_cache, record):
+def _check_video_abi_compat(apt_cache, package):
     xorg_video_abi = None
 
-    # determine current X.org video driver ABI
-    try:
-        for p in apt_cache['xserver-xorg-core'].candidate.provides:
-            if p.startswith('xorg-video-abi-'):
-                xorg_video_abi = p
-                # logging.debug('_check_video_abi_compat(): Current X.org video abi: %s', xorg_video_abi)
-                break
-    except (AttributeError, KeyError):
-        logging.debug('_check_video_abi_compat(): xserver-xorg-core not available, cannot check ABI')
+    if package.name.startswith('nvidia-driver-'):
+        xorg_driver_name = package.name.replace('nvidia-driver-', 'xserver-xorg-video-nvidia-')
+        try:
+            package = apt_cache.__getitem__(xorg_driver_name)
+        except KeyError:
+            logging.debug('Cannot find %s package in the cache. Cannot check ABI' % (xorg_driver_name))
+            return True
+
+    records = apt_pkg.PackageRecords(apt_cache)
+    depcache = apt_pkg.DepCache(apt_cache)
+    candidate = depcache.get_candidate_ver(package)
+    records.lookup(candidate.file_list[0])
+    deps = records.__getitem__('Depends').split(', ')
```

Comment:
Use `candidate.depends_list` or `candidate.depends_list_str` and lose the record parsing.